addon-test-support/setup-test.js

```diff
@@ -3,16 +3,95 @@
  */
 
 import {classify} from '@ember/string'
-import {setupModelTest, setupTest} from 'ember-mocha'
+import {setupTest} from 'ember-mocha'
 
 import './typedefs'
 
 // Workaround to allow stubbing dependencies during testing
 export const deps = {
-  setupModelTest,
   setupTest
 }
 
+/**
+ * Create test subject for Module type
+ * @param {String} name - the name of the Module
+ * @returns {Test} test context object with a subject method
+ */
+export function createTestContextForModule (name) {
+  return {
+    subject (options = {}, nameOverrides) {
+      if (!this.owner) {
+        console.error('Make sure you have run setup() before trying to access subject.')
+      }
+
+      const subject = this.owner.lookup(nameOverrides || name)
+
+      Object.keys(options).forEach(key => {
+        subject.set(key, options[key])
+      })
+
+      return subject
+    }
+  }
+}
+
+/**
+ * Create test subject for Model type
+ * @param {String} name - the name of the Model
+ * @returns {Test} test context object with a subject method
+ */
+export function createTestContextForModel (name) {
+  return {
+    subject (options = {}, nameOverrides) {
+      if (!this.owner) {
+        console.error('Make sure you have run setup() before trying to access subject.')
+      }
+
+      const subject = this.owner.lookup('service:store').createRecord(nameOverrides || name, options)
+
+      return subject
+    }
+  }
+}
+
+/**
+ * Create test subject for Serializer type
+ * @param {String} name - the name of the Serializer
+ * @returns {Test} test context object with a subject method
+ */
+export function createTestContextForSerializer (name) {
+  return {
+    subject (options = {}, nameOverrides) {
+      if (!this.owner) {
+        console.error('Make sure you have run setup() before trying to access subject.')
+      }
+
+      this.owner.lookup('service:store').createRecord(nameOverrides || name, options)
```

Comment:
Do we need this?

Reply:
Yup, testing methods on Serializer and Adapter will easily trigger a record/model lookup under the hook. And the test will error out when there's no corresponding record.